docs/guides/integration.md

```diff
@@ -0,0 +1,423 @@
+
+For simplicity's sake, this guide will use a scala application with a very simple set of dependencies.  In principle, these steps will work with any service whos dependencies can be specified in a docker-compose file. 
+
+## The Example App
+
+The app has one purpose, it returns the first 5 countries alphabetically via standard out.  To do this it connects to a database through a data access layer and returns the results from a Postgres table.  The specifics of the app are not relevant to the testing method but are outlined here for clarity.
+
+The app has unit tests that don't require any dependencies and integration tests that exercise the data access layer and the connection to the database. 
+
+{% method %}
+{% sample lang="App" %}
+Application code:
+```scala
+Object Main extends App {
+  val dal = new DataAccessLayer()
+  val dv = new DataVersion()
+
+  if(dv.version() > 1)
+  {
+    implicit val cs = IO.contextShift(ExecutionContext.global)
+    val xa = Transactor.fromDriverManager[IO](
+      "org.postgresql.Driver", 
+      "jdbc:postgresql://localhost:5432/iso3166", 
+      "postgres",
+      "postgres"
+    )
+
+    val countries = dal.countries(5)
+                       .transact(xa).unsafeRunSync
+                       .toList.map(_.name).mkString(", ")
+
+    println(s"The first 5 countries alphabetically are: $countries")
+  }
+}
+```
+The output of running the application:
+
+``` bash
+> sbt run
+The first 5 countries are Afghanistan, Albania, Algeria, American Samoa, Andorra
+```
+
+{% sample lang="Unit Test" %}
+``` scala
+class DataVersionSpec extends FlatSpec {
+
+  val dv = new DataVersion()
+  "Data Version " should " be positive" in {
+    assert(dv.version > 0)
+  }
+}
+```
+Output of running unit tests:
+``` bash
+> sbt test
+[info] DataVersionSpec:
+[info] Data Version
+[info] - should be positive
+[info] Run completed in 810 milliseconds.
+[info] Total number of tests run: 1
+
+```
+
+{% sample lang="Integration Test" %}
+Integration test:
+``` scala
+class DatabaseIntegrationTest extends FlatSpec {
+  implicit val cs = IO.contextShift(ExecutionContext.global)
+
+  val xa = Transactor.fromDriverManager[IO](
+    "org.postgresql.Driver", 
+    "jdbc:postgresql://localhost:5432/iso3166", 
+    "postgres",
+    "postgres"
+  )
+
+  "An table" should "have country data" in {
+    val dal = new DataAccessLayer()
+    assert(dal.countries(5).transact(xa).unsafeRunSync.size == 5)
+  }
+}
+```
+Ouput:
+``` bash
+>sbt it:test
+[info] DatabaseIntegrationTest:
+[info] An table
+[info] - should have country data
+[info] Run completed in 2 seconds, 954 milliseconds.
+[info] Total number of tests run: 1
+```
+{% endmethod %}
+
+## The Earthfile
+
+We start with a simple earthly file that can build and create a docker file for our app.  See [Basic]() guide for more details on that.
```

Comment:
Missing link here.

Note that there are multiple places below with missing links.